include/odp/api/spec/schedule.h

```diff
@@ -368,6 +368,26 @@ void odp_schedule_order_lock(unsigned lock_index);
 void odp_schedule_order_unlock(unsigned lock_index);
 
 /**
+ * Release ordered context lock and acquires new lock
+ *
+ * This call is valid only when holding an ordered synchronization context.
+ * Release a previously locked ordered context lock and acquires
+ * a new ordered context lock.
+ * This call is valid only when there is a single ordered context lock and
+ * is invalid if there are nested ordered context lock.
```

Comment:
You should also clarify what is meant by nested ordered context lock here. As I recall, the restriction is that the caller may only hold a single ordered lock because you didn't want to include the index of the lock to be released as part of this call. So that restriction should be stated explicitly with cautions that results are undefined if it is not observed.

Is the reason for this restriction that your HW doesn't support holding multiple ordered locks concurrently or is there some restriction in the lock switching mechanism that prevents this?